credentials/tls.go

```diff
@@ -50,6 +51,22 @@ func (t TLSInfo) AuthType() string {
 	return "tls"
 }
 
+// ValidateAuthority validates that the provided authority being used to
+// override the :authority header is valid by verifying it against the peer
+// certificates. It returns nil on successful validation, or a non-nil error if
+// the validation fails.
+func (t TLSInfo) ValidateAuthority(authority string) error {
+	var errs []error
+	for _, cert := range t.State.PeerCertificates {
+		var err error
+		if err = cert.VerifyHostname(authority); err == nil {
+			return nil
+		}
+		errs = append(errs, err)
+	}
+	return fmt.Errorf("credentials: invalid authority %q: %w", authority, errors.Join(errs...))
```

Comment:
Embedding the error from the tls package using the `%w` format string means that we are making that error a part of our API. I don't think we want that.

Please read this section: go/go-style/best-practices#error-extra-info